framework/src/modules/chain/transport.js

```diff
@@ -423,19 +423,30 @@ class Transport {
 			 * @todo Add @returns tag
 			 * @todo Add description of the function
 			 */
-			postSignature(query, cb) {
-				__private.receiveSignature(query.signature, (err, code) => {
-					if (err) {
-						return setImmediate(cb, null, {
-							success: false,
-							code,
-							errors: err,
-						});
-					}
-					return setImmediate(cb, null, {
-						success: true,
-					});
-				});
+			async postSignature(query) {
+				const valid = library.schema.validate(
+					query.signature,
+					definitions.Signature
+				);
+
+				if (!valid) {
+					const err = library.schema.getLastErrors();
+					const error = new TransactionError(err[0].message);
+					return {
+						success: false,
+						code: 400,
+						errors: [error],
+					};
+				}
+
+				return modules.transactionPool
```

Comment:
why no await here?

Reply:
Because this function doesn't throw anything.
Instead, it always return an object.